Fatras/include/ActsFatras/Kernel/InteractionList.hpp

```diff
@@ -0,0 +1,344 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2018-2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Material/MaterialSlab.hpp"
+#include "ActsFatras/EventData/Particle.hpp"
+
+#include <bitset>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+namespace ActsFatras {
+namespace detail {
+
+/// Retrieve index of the first matching type in the tuple.
+///
+/// Taken from https://stackoverflow.com/a/18063608.
+template <class T, class Tuple>
+struct TupleIndexOf;
+template <class T, class... Types>
+struct TupleIndexOf<T, std::tuple<T, Types...>> {
+  static constexpr std::size_t value = 0u;
+};
+template <class T, class U, class... Types>
+struct TupleIndexOf<T, std::tuple<U, Types...>> {
+  static constexpr std::size_t value =
+      1u + TupleIndexOf<T, std::tuple<Types...>>::value;
+};
```

Comment:
Would you mind adding a small unit test for this?

Reply:
Will do.